airbyte-lib/airbyte_lib/caches/base.py

```diff
@@ -763,6 +792,93 @@ def _table_exists(
         """Return true if the given table exists."""
         return table_name in self._get_tables_list()
 
+    def _ensure_internal_tables(self) -> None:
+        self._ensure_schema_exists()
+        engine = self.get_sql_engine()
+        Base.metadata.create_all(engine)
+
+    @overrides
+    def register_source(
+        self,
+        source_name: str,
+        incoming_source_catalog: ConfiguredAirbyteCatalog,
+    ) -> None:
```

Comment:
Noted also where this is called (quoted comment text below): I think it is worthwhile to filter the records we store in the metadata table down to those that we expect to receive data. One way to do that is to accept an optional stream_names filter in this method signature.

> Because sources can have a hundred or more streams, wdyt of passing the list of _selected_ streams (or all/None if nothing is selected). Since this registration occurs during "read()", we should know here which of the streams are actually going to be used and which will be ignored.
> 
> If we can implement without too much added complexity, I'd rather not store metadata on streams that are not going to receive data.
> 

 Wdyt?

Reply:
Linked the place where this is happening in the other comment.